scheduler/src/cook/rest/api.clj

```diff
@@ -954,12 +978,27 @@
                (not (contains? (get-gpu-models-on-pool (config/valid-gpu-models) pool-name) requested-gpu-model)))
       (throw (ex-info (str "The following GPU model is not supported: " requested-gpu-model) {})))))
 
+(defn validate-disk-job
+  "Validates that a job requesting disk is satisfying the following conditions:
+    - User can request size but not type; User can request both size and type; User cannot request only type and not size
+    - Requested type must be a valid type
+    - TODO: validate that requested size is less than a certain amount
```

Comment:
This is done by the schema validation, right?

Reply:
yep, will remove